GVFS/GVFS.Platform.Windows/WindowsFileSystem.cs

```diff
@@ -117,6 +117,57 @@ public bool IsSocket(string fileName)
             return false;
         }
 
+        /// <summary>
+        /// Creates the specified directory (and its ancestors) if they do not
+        /// already exist.
+        ///
+        /// If the specified directory does not exist this method:
+        ///
+        ///  - Creates the directory and its ancestors
+        ///  - Adjusts the ACLs of 'directoryPath' (the ancestors' ACLs are not
+        ///    modified).
+        /// </summary>
+        /// <param name="directoryPath"></param>
+        /// <remarks>
+        /// The following permissions are typically present on deskop and missing on Server.
+        /// These are the permissions added by this method.
+        ///
+        ///   ACCESS_ALLOWED_ACE_TYPE: NT AUTHORITY\Authenticated Users
+        ///          [OBJECT_INHERIT_ACE]
+        ///          [CONTAINER_INHERIT_ACE]
+        ///          [INHERIT_ONLY_ACE]
+        ///        DELETE
+        ///        GENERIC_EXECUTE
+        ///        GENERIC_WRITE
+        ///        GENERIC_READ
+        /// </remarks>
+        public void CreateDirectoryAccessibleByAuthUsers(string directoryPath)

```

Comment:
Do we need to do any exception handling in this method or is it always expected to always succeed?

Reply:
Thanks for pointing this out, based on how/where the callers are calling this method I think it makes sense to convert it to a `Try` method (rather than forcing them to have better exception handling added).

I will update this.